jsapi/src/main.js

```diff
@@ -437,6 +450,138 @@ lux.Map.prototype.addSearch = function(target) {
 };
 
 /**
+ * Adds a GPX file on the map
+ * @param {string} url Url to the GPX file
+ * @export
+ */
+lux.Map.prototype.addGPX = function(url) {
+  var style = {
+    'Point': new ol.style.Style({
+      image: new ol.style.Circle({
+        fill: new ol.style.Fill({
+          color: 'rgba(255,255,0,0.4)'
+        }),
+        radius: 5,
+        stroke: new ol.style.Stroke({
+          color: '#ff0',
+          width: 1
+        })
+      })
+    }),
+    'LineString': new ol.style.Style({
+      stroke: new ol.style.Stroke({
+        color: '#f00',
+        width: 3
+      })
+    }),
+    'MultiLineString': new ol.style.Style({
+      stroke: new ol.style.Stroke({
+        color: '#f00',
+        width: 3
+      })
+    })
+  };
+
+  var styleFunction = function(feature) {
+    return style[feature.getGeometry().getType()];
+  };
+
+  this.addVector(url, new ol.format.GPX(), styleFunction);
+};
+
+/**
+ * Adds a KML file on the map
+ * @param {string} url Url to the GPX file
+ * @export
+ */
+lux.Map.prototype.addKML = function(url) {
+  this.addVector(url, new ol.format.KML());
+};
+
+/**
+ * Adds a KML file on the map
+ * @param {string} url Url to the vector file
+ * @param {ol.format.GPX|ol.format.KML} format The format.
+ * @param {ol.style.StyleFunction=} opt_styleFunction The style function.
+ * @export
+ */
+lux.Map.prototype.addVector = function(url, format, opt_styleFunction) {
+  var options = {
+    source: new ol.source.Vector({
+      url: url,
+      format: format
+    })
+  };
+  var popup;
+
+  if (opt_styleFunction) {
+    options.style = opt_styleFunction;
+  }
+  this.promise.then(function() {
+    var vector = new ol.layer.Vector(options);
+    this.addLayer(vector);
+
+    ol.events.listen(vector.getSource(), ol.source.VectorEventType.ADDFEATURE,
+        function() {
+          var size = /** {ol.Size} */ (this.getSize());
+          goog.asserts.assert(size !== undefined, 'size should be defined');
```

Comment:
It can help the integrator to better understand the reason why the extent is not changed.